Sprint_14

BezrukovAndrej opened the pull request:
Исправлены все ошибки:
1) если список трекеров для отображения пуст (в том числе — как результат поиска и/или выбора даты), то появляется заглушка. 
2)  при нажатии на + изображение кнопки меняется на галочку «выполнено для указанной даты», а счётчик увеличивается. 
3)  нельзя отметить карточку для будущей даты.
4)  структуры, Tracker, TrackerCategory и TrackerRecord, сделаны без мутабельных свойств, все их поля отмечены let.

BezrukovAndrej committed:
Согласно 14 спринту
Исправлены все ошибки:
1) если список трекеров для отображения пуст (в том числе — как результат поиска и/или выбора даты), то появляется заглушка.
2)  при нажатии на + изображение кнопки меняется на галочку «выполнено для указанной даты», а счётчик увеличивается.
3)  нельзя отметить карточку для будущей даты.
4)  структуры, Tracker, TrackerCategory и TrackerRecord, сделаны без мутабельных свойств, все их поля отмечены let.

BezrukovAndrej committed:
Исправлены все ошибки:

если список трекеров для отображения пуст (в том числе — как результат поиска и/или выбора даты), то появляется заглушка.
при нажатии на + изображение кнопки меняется на галочку «выполнено для указанной даты», а счётчик увеличивается.
нельзя отметить карточку для будущей даты.
структуры, Tracker, TrackerCategory и TrackerRecord, сделаны без мутабельных свойств, все их поля отмечены let.

helecta commented on Tracker/ViewControllers /NewTrackerViewController.swift:
```diff
@@ -0,0 +1,267 @@
+    @objc private func createButtonTapped() {
+        dismiss(animated: true)
+        let trackerName = textField.text ?? ""
+        self.delegate?.addNewTrackerCategory(TrackerCategory(title: "Новая категория", trackers: [Tracker.init(id: UUID(), text: trackerName, emoji: "🔥", color: .uiBlue, schedule: self.schedule)]))
```
Лучше такие длинные методы разбивать строчками и выносить в переменные:
```
        let mockTracker = Tracker(id: UUID(),
                                  text: trackerName,
                                  emoji: "🔥",
                                  color: .uiBlue,
                                  schedule: schedule)
        let mockCategory = TrackerCategory(title: "Новая категория",trackers: [mockTracker])
        delegate?.addNewTrackerCategory(mockCategory)
```
Так код значительно легче и удобней читать

helecta replied:
И тут `self` необязательно использовать, поэтому лучше его убрать.
Подробнее можешь почитать в [код стайле](https://github.com/tutu-ru/swift-style-guide#комментарии)

BezrukovAndrej committed:
Исправлено:
1) На экране создания привычки, после выбора расписания, выбор не отображается на экране, при этом можно создать трекер.

Не исправлено и согласно чек листу отсутствует данный функционал
1) 1. Нет возможности отметить карточку для прошедшей даты.

П.C
На экране создания привычки, после выбора расписания, выбор не отображается на экране, при этом можно создать трекер.
Данного пункта нету согласно чек листу!!

BezrukovAndrej committed:
убрал self
